draft-ietf-httpbis-resumable-upload.md

```diff
@@ -512,6 +512,36 @@ If the client knows the integrity digest of the content from an upload creation
 
 The server might process the uploaded data and make its results available in another resource during or after the upload. This subsequent resource is different from the upload resource created by the upload creation request ({{upload-creation}}). The subsequent resource does not handle the upload process itself, but instead facilitates further interaction with the uploaded data. The server MAY indicate the location of this subsequent resource by including the `Content-Location` header field in the informational or final responses generated while creating ({{upload-creation}}), appending to ({{upload-appending}}), or retrieving the offset ({{offset-retrieving}}) of an upload. For example, a subsequent resource could allow the client to fetch information extracted from the uploaded data.
 
+# Upload Strategies
+
+The definition of the upload creation request ({{upload-creation}}) provides the client with flexibility to choose whether the file is fully or partially transferred in the first request, or if no file data is included at all. Which behavior is best largely depends on the client's capabilities, its intention to avoid data re-transmission, and its knowledge about the server's support for resumable uploads.
+
+The following subsections describe two typical upload strategies that are suited for common environments. Note that these modes are never explicitly communicated to the server and clients are not required to stick to one strategy, but can mix and adapt them to their needs.
+
+## Optimistic Upload Creation
+
+An "optimistic upload creation" can be used if the client assumes that the server supports resumable uploads under the target URI and that the client is capable of handling and processing interim responses. The upload creation request then includes the full file because the client anticipates that the file will be transferred without interruptions or resumed if an interruption occurs.
```

Comment:
> assumes that the server supports resumable uploads under the target URI

This seems to be a bit in conflict with the "Upgrading To Resumable Uploads" subsection now that they are merged. Maybe we can say "assumes that the server supports resumable uploads under the target URI, or has no knowledge of the server support", but then that sentence doesn't really say anything.

"Careful Upload Creation" is where the server must support resumable uploads. "Optimistic Upload Creation" is OK if the server doesn't.

Reply:
Yes, the client's knowledge about the server support was a difference of the two sections at first. Now that they are merged, the text gets a bit more unwieldy, but a added https://github.com/httpwg/http-extensions/commit/de5992146b5957168dc2c7e0af2a2dbe64086ab5 to correct the statements about the client's knowledge. What do you think? Is that sufficient or do we need more significant changes?

Reply:
Looks good, thanks